src/client.js

```diff
@@ -464,35 +487,33 @@ export default class StitchClient {
 
     let { url, fetchArgs } = this._fetchArgs(resource, method, options);
     if (!options.noAuth) {
-      return this.authedId().then(authedId => {
-        if (!this.authedId) {
-          return Promise.reject(new StitchError('Must auth first', ErrUnauthorized));
-        }
+      if (!this.authedId()) {
+        return Promise.reject(new StitchError('Must auth first', ErrUnauthorized));
+      }
 
-        let tokenPromise =
-          options.useRefreshToken ? this.auth.getRefreshToken() : this.auth.getAccessToken();
-
-        // If access token is expired, proactively get a new one
-        if (!options.useRefreshToken) {
-          return this.auth.isAccessTokenExpired().then(isAccessTokenExpired => {
-            if (isAccessTokenExpired) {
-              return this.auth.refreshToken().then(() => {
-                options.refreshOnFailure = false;
-                return this._do(resource, method, options);
-              });
-            }
-
-            return tokenPromise.then(token => {
-              fetchArgs.headers.Authorization = `Bearer ${token}`;
-              return this._fetch(url, fetchArgs, resource, method, options);
+      let tokenPromise =
+        options.useRefreshToken ? this.auth.getRefreshToken() : this.auth.getAccessToken();
+
+      // If access token is expired, proactively get a new one
+      if (!options.useRefreshToken) {
+        return this.auth.isAccessTokenExpired().then(isAccessTokenExpired => {
```

Comment:
[q] `isAccessTokenExpired` cool be synchronous now too I think, right?

Reply:
👍